src/assembler/instruction.h

```diff
@@ -289,14 +291,51 @@ class Instruction {
     }
 
     AssembleRes assemble(const Assembler::TokenizedSrcLine& line) const {
+        QString Hint = "";
+        // if(dynamic_cast<Imm*>(m_fields[1].get()))
+        //     std::cout << "Its Imm" << std::endl;
+        // else if(dynamic_cast<Reg*>(m_fields[1].get()))
+        //     std::cout << "Its Reg" << std::endl;
+        // else
+        //     std::cout << "Nothing" << std::endl;
+
+        for(int i = 0; i < (m_expectedTokens-1); i++){
+            //Imm
+            if(dynamic_cast<Imm*>(m_fields[i].get()))
```

Comment:
This if branch can be transformed to
```
if (auto* immField = dynamic_cast<Imm*>(m_fields[i].get())) {
    ...
}
```
And from this, i think you should also include the allowed width of the immediate value, as in your previous proposal.